sycl/doc/SYCLGraphUsageGuide.md

```diff
@@ -0,0 +1,493 @@
+# SYCL-Graph Usage Guide and Examples
```

Comment:
Should we use a hyphen here? I think we use 'SYCL Graph', 'SYCL-Graph', and 'SYCL Graphs' in various places, but 'SYCL Graph' is most common.